#!/bin/sh /usr/share/dpatch/dpatch-run
## 30_conf.d_support_builtin.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Support Debian conf.d style out-of-the-box without perl hackery
## DP: in the main legacy config file.

@DPATCH@
diff -urNad amavisd-new~/amavisd amavisd-new/amavisd
--- amavisd-new~/amavisd	2007-09-17 22:29:49.000000000 +0200
+++ amavisd-new/amavisd	2007-09-17 22:31:56.000000000 +0200
@@ -2394,6 +2394,25 @@
     Amavis::Util::read_text("$dir/template-spam-admin.txt", $file_chset);
 }
 
+# attempt to read a list of config files to use instead of the default
+# one, through an external help script. Used by the Debian distribution.
+sub find_config_files(@) {
+  my(@dirs) = @_;
+  my(@config_files);
+
+# my($old_path) = $ENV{PATH};
+# $ENV{PATH} = "/bin:/usr/bin";
+  foreach my $dir (@dirs) {
+    push(@config_files, `run-parts --list "$dir"`);
+  }
+# $ENV{PATH} = $old_path;
+
+  # untaint and chomp.  This data is secure as we check the files
+  # themselves later
+  chomp(@config_files);
+  map { untaint($_) } @config_files;
+}
+
 #use CDB_File;
 #sub tie_hash($$) {
 # my($hashref, $filename) = @_;
@@ -11054,7 +11073,10 @@
   Amavis::Lookup::RE->new(@$Amavis::Conf::map_full_type_to_short_type_re);
 
 # default location of the config file if none specified
-push(@config_files, '/etc/amavisd.conf')  if !@config_files;
+if (!@config_files) {
+  @config_files = Amavis::Util::find_config_files('/usr/share/amavis/conf.d',
+                                                       '/etc/amavis/conf.d');
+}
 # Read and evaluate config files, which may override default settings
 Amavis::Conf::include_config_files(@config_files);
 Amavis::Conf::supply_after_defaults();
